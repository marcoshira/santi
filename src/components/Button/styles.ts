import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
    background: inherit;
    color: ${theme.colors.teal};
    /* border: 0.5px solid ${theme.colors.teal}; */
    border: none;
    font-size: ${theme.font.sizes.ltmedium};
    font-weight: 500;
    margin-left: ${theme.spacings.xxsmall};
    margin-top: 5px;
    margin-bottom: 4px;
    position: relative;
    transition: all 300ms ease-in-out;

    &:disabled {
      background: ${theme.colors.mediumGrey};
      color: ${theme.colors.white};
      border: 2px solid ${theme.colors.mediumGrey};
      transition: none;

      &:hover {
        background: ${theme.colors.mediumGrey};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.mediumGrey};
        cursor: not-allowed;
      }

      &::after {
        content: none;
      }
    }
    &:hover {
      color: ${theme.colors.thirdColor};
      cursor: pointer;
    }

    &::after {
      content: '';
      position: absolute;
      bottom: 0.2rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.thirdColor};
      transition: all 200ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
