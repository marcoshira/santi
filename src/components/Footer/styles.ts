import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.2rem solid ${theme.colors.mediumGrey};

    a {
      color: inherit;
      text-decoration: none;
    }
  `}
`;
