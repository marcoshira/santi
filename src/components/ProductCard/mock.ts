import { ProductCardArrayProps } from '../../shared-types/shared-types';

export default {
  id: '3',
  attributes: {
    name: 'Breath of the Wild',
    description:
      'Entre num mundo de aventuras Esqueça tudo o que sabe sobre os jogos The legend of zelda. Entre num mundo de descoberta, exploração e aventura em The legend of zelda: Breath of the Wild, o novo título que quebra com padrões da série. Atravesse campos, florestas e montanhas para descobrir o que aconteceu ao reino arruinado de Hyrule nesta deslumbrante aventura ao ar livre. Explora Hyrule com total liberdade Escale torres e picos de montanhas em busca de novos destinos e defina seu próprio caminho para chegar lá e entrar na floresta. Enquanto isso combata inimigos gigantescos, cace bestas selvagens e reúna ingredientes para os alimentos e as poções que precisará para sobreviver nesta missão. Mais de 100 Shrines of Trials para descobrir e explorar Há santuários espalhados por todo o mundo de Hyrule, santuários esses que terá que descobrir e na ordem que quiser. Procure-os de diversas maneiras e em cada resolva uma variedade de quebra-cabeças.',
    price: 29999,
    stock: 29,
    cover: {
      data: {
        attributes: {
          url: 'https://res.cloudinary.com/djqrityvi/image/upload/v1652741206/botw_f71c3467c2.jpg',
        },
      },
    },
    product_type: {
      data: {
        id: '3',
        attributes: {
          title: 'Switch Game',
        },
      },
    },
  },
} as ProductCardArrayProps;
