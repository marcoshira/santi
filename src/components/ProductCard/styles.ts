import styled, { css } from 'styled-components';

export const Wrapper = styled.a`
  ${({ theme }) => css`
    text-decoration: none;
    color: ${theme.colors.navy};
    cursor: pointer;
    min-width: 30rem;
    max-width: 40rem;

    @media ${theme.media.lteSmall} {
      max-width: 30rem;
      min-width: auto;
    }
  `}
`;

export const CardWrapper = styled.div`
  ${({ theme }) => css`
    /* border: 2px solid ${theme.colors.teal}; */
    box-shadow: 3px 3px 15px ${theme.colors.teal};
    border-radius: 15px;
    min-height: 35rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    position: relative;
    background: ${theme.colors.footerWhite};
    transition: all 300ms ease-in-out;
    &:hover {
      opacity: 0.8;

      p {
        color: ${theme.colors.thirdColor};
      }
    }

    @media ${theme.media.lteSmall} {
      &:hover {
        opacity: 1;
      }
    }
  `}
`;

export const ImageWrapper = styled.figure`
  ${({ theme }) => css`
    margin-top: 1rem;
    display: flex;
    min-height: 26rem;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    img {
      max-height: 90%;
      max-width: 90%;
      transition: all 300ms ease-in-out;

      @media ${theme.media.lteSmall} {
        &:hover {
          opacity: 0.8;
        }
      }
    }
  `}
`;

export const TextWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    position: absolute;
    bottom: 2rem;

    p {
      margin: 1rem 0 0 0;
      font-size: 1.8rem;
      transition: all 300ms ease-in-out;
    }
  `}
`;
