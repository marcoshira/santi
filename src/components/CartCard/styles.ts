import styled, { css } from 'styled-components';
import { Wrapper as Btn } from '../Button/styles';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    width: 100%;
    overflow-x: hidden;
    box-shadow: 3px 3px 15px ${theme.colors.teal};
    border-radius: 15px;
    color: ${theme.colors.navy};
    background: ${theme.colors.footerWhite};
    position: relative;
    padding: ${theme.spacings.small} 0;
    margin-bottom: ${theme.spacings.ltmedium};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      min-height: 50rem;
      width: 85%;
      justify-content: center;
      align-items: center;
      margin: ${theme.spacings.medium} auto;
    }

    a {
      cursor: pointer;
      &:hover {
        img {
          opacity: 0.6;
        }
      }
    }

    img {
      border-radius: 10px;
      height: 20rem;
      transition: ${theme.transitions.faster};

      @media ${theme.media.lteMedium} {
        height: 15rem;
      }
    }

    ${Title} {
      margin: 0;
    }
  `}
`;

export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    width: 55%;
    a {
      text-decoration: none;
      color: ${theme.colors.navy};
      transition: ${theme.transitions.faster};
      &:hover {
        color: ${theme.colors.thirdColor};
      }
    }

    @media ${theme.media.lteMedium} {
      width: 100%;
      padding: ${theme.spacings.medium};
    }
  `}
`;

export const ImageWrapper = styled.div`
  display: flex;
  justify-content: center;
  width: 45%;
`;

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    flex-direction: row;
    margin-top: 2rem;
    margin-right: 2rem;

    ${Btn} {
      min-width: 48%;

      margin: 0rem;
      margin-right: 1rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.footerWhite};
      color: ${theme.colors.teal};

      &:hover {
        border: 2px solid ${theme.colors.thirdColor};
        background: ${theme.colors.footerWhite};
        color: ${theme.colors.thirdColor};
      }

      &::after {
        content: none;
      }

      :nth-of-type(2) {
        background: ${theme.colors.teal};
        color: ${theme.colors.footerWhite};

        &:hover {
          border: 2px solid ${theme.colors.teal};
          background: ${theme.colors.hoverTeal};
        }
      }
    }

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      margin-right: 0rem;

      ${Btn} {
        width: 100%;
        margin: ${theme.spacings.tiny} 0;
      }
    }
  `}
`;
