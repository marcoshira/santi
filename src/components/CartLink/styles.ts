import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: absolute;
    left: 40rem;
    top: 9rem;
    width: calc(100vw - 55rem);

    @media ${theme.media.lteMedium} {
      width: 100%;
      left: auto;
      top: 6rem;
      margin: 0 auto;

      > ${Title} {
        margin-left: ${theme.spacings.ltlarge};
      }
    }
  `}
`;
