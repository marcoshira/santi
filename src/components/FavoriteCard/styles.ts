import styled, { css } from 'styled-components';
import { Wrapper as Btn } from '../Button/styles';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    width: 100%;
    box-shadow: 3px 3px 15px ${theme.colors.teal};
    border-radius: 15px;
    color: ${theme.colors.navy};
    background: ${theme.colors.footerWhite};
    position: relative;
    padding: ${theme.spacings.small} 0;
    margin-bottom: ${theme.spacings.ltmedium};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      min-height: 53rem;
      width: 85%;
      justify-content: center;
      align-items: center;
      margin: ${theme.spacings.medium} auto;
      padding-bottom: 0;
    }

    a {
      cursor: pointer;
      &:hover {
        img {
          opacity: 0.6;
        }
      }
    }

    img {
      border-radius: 10px;
      height: 20rem;
      transition: ${theme.transitions.faster};

      @media ${theme.media.lteMedium} {
        height: 15rem;
      }
    }

    ${Title} {
      margin: 0;
    }
  `}
`;

export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    width: 55%;
    a {
      text-decoration: none;
      color: ${theme.colors.navy};
      transition: ${theme.transitions.faster};
      &:hover {
        color: ${theme.colors.thirdColor};
      }
    }
    svg {
      height: 3rem;
      position: absolute;
      top: 2rem;
      right: 2rem;
      color: ${theme.colors.teal};
      cursor: pointer;
      transition: ${theme.transitions.fast};

      &:hover {
        color: ${theme.colors.hoverTeal};
      }
    }

    @media ${theme.media.lteMedium} {
      width: 100%;
      padding: ${theme.spacings.medium};
    }
  `}
`;

export const ImageWrapper = styled.div`
  ${({ theme }) => css`
    width: 45%;
    display: flex;
    justify-content: center;

    img {
    }
  `}
`;

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-around;
    flex-direction: row;
    margin-top: 3rem;
    ${Btn} {
      min-width: 47%;

      margin: 0rem;
      margin-right: 1rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.footerWhite};
      color: ${theme.colors.teal};

      &:hover {
        border: 2px solid ${theme.colors.thirdColor};
        background: ${theme.colors.footerWhite};
        color: ${theme.colors.thirdColor};
      }

      &::after {
        content: none;
      }

      :nth-of-type(2) {
        background: ${theme.colors.teal};
        color: ${theme.colors.footerWhite};

        &:hover {
          border: 2px solid ${theme.colors.teal};
          background: ${theme.colors.hoverTeal};
        }
      }
    }

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      margin-right: 0rem;

      ${Btn} {
        width: 100%;
        margin: ${theme.spacings.tiny} 0;
      }
    }
  `}
`;

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 50%;
    justify-content: left;
    flex-flow: row nowrap;
    font-size: 1.6rem;
    margin-top: 3rem;

    @media ${theme.media.lteMedium} {
      width: 100%;
      margin-top: 1rem;
    }

    label {
      margin: 8px ${theme.spacings.ltlarge} 0 0;
      color: ${theme.colors.darkGrey};

      @media ${theme.media.lteMedium} {
        margin: 7px ${theme.spacings.hero} 0 0;
      }
    }

    input {
      width: 40px;
      display: flex;
      text-align: center;
      align-items: center;
      justify-content: center;
      border: 1px solid ${theme.colors.grey};
      height: 35px;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    padding: 0 10px;
    background: ${theme.colors.footerWhite};
    border: 1px solid ${theme.colors.grey};
  `}
`;
