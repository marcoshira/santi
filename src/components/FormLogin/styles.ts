import styled, { css } from 'styled-components';
import { Wrapper as Button } from '../Button/styles';

export const Wrapper = styled.form``;

export const ButtonWrapper = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.medium};
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    ${Button} {
      a {
        text-decoration: none;
        color: ${theme.colors.footerWhite};
      }
      width: 15rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.teal};
      color: ${theme.colors.footerWhite};
      border-radius: ${theme.spacings.tiny};
      margin-left: 0;
      &:hover {
        background: ${theme.colors.hoverTeal};
      }
      &::after {
        height: 0;
      }
      &:disabled {
        background: ${theme.colors.mediumGrey};
        color: ${theme.colors.footerWhite};
        border: 2px solid ${theme.colors.mediumGrey};

        &:hover {
          background: ${theme.colors.mediumGrey};
          color: ${theme.colors.white};
          border: 2px solid ${theme.colors.mediumGrey};
          cursor: not-allowed;
        }

        &::after {
          content: none;
        }
      }

      @media ${theme.media.lteSmall} {
        width: 100%;
        padding: ${theme.spacings.xsmall} ${theme.spacings.xsmall};
      }
    }
  `}
`;
