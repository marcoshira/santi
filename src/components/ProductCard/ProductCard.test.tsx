import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ProductCard } from '.';
import { ProductCardProps } from '../../shared-types/shared-types';
import mock from './mock';

const props: ProductCardProps = mock;

describe('<ProductCard />', () => {
  it('should render', () => {
    const { container } = renderTheme(<ProductCard {...props} />);

    expect(screen.getByText(mock.attributes.name)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
