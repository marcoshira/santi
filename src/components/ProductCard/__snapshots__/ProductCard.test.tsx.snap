// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0A1128;
  cursor: pointer;
  min-width: 30rem;
  max-width: 40rem;
}

.c1 {
  box-shadow: 3px 3px 15px #8c4bff;
  border-radius: 15px;
  min-height: 35rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  background: #FFFFFF;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover {
  opacity: 0.8;
}

.c1:hover p {
  color: #ff4a4a;
}

.c2 {
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 26rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 img {
  max-height: 90%;
  max-width: 90%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  bottom: 2rem;
}

.c3 p {
  margin: 1rem 0 0 0;
  font-size: 1.8rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

@media (max-width:440px) {
  .c0 {
    max-width: 30rem;
    min-width: auto;
  }
}

@media (max-width:440px) {
  .c1:hover {
    opacity: 1;
  }
}

@media (max-width:440px) {
  .c2 img:hover {
    opacity: 0.8;
  }
}

<div>
  <a
    class="c0"
    href="/products/2"
  >
    <div
      class="c1"
    >
      <figure
        class="c2"
      >
        <img
          src="https://res.cloudinary.com/djqrityvi/image/upload/v1652740674/ps5_8a252e1eaa.jpg"
        />
      </figure>
      <div
        class="c3"
      >
        <p>
          PlayStation 5
        </p>
        <p>
          R$ 
          4799.99
        </p>
      </div>
    </div>
  </a>
</div>
`;
